projects/packages/forms/src/dashboard/components/layout/style.scss

```diff
@@ -32,6 +32,13 @@ body.toplevel_page_jetpack-forms {
 
 #jp-forms-dashboard .jp-forms__layout {
 
+	.jp-forms__layout-header {
+		display: flex;
```

Comment:
HStack component might work here to move some styling into componentry, but not a biggie either way.